File: sdk/js/examples/transaction.ts
// Transaction example for MoltChain SDK

import { Connection, PublicKey, TransactionBuilder } from '../dist/index';

async function main() {
  const connection = new Connection('http://localhost:8899');

  console.log('🦞 MoltChain Transaction Example\n');

  // Create a transfer transaction
  const from = new PublicKey('FromPublicKeyHere...');
  const to = new PublicKey('ToPublicKeyHere...');
  const amount = 1_000_000_000; // 1 MOLT

  console.log('Building transaction...');
  
  // Get recent blockhash
  const latestBlock = await connection.getLatestBlock();
  
  // Build transaction
  const transaction = new TransactionBuilder()
    .add(TransactionBuilder.transfer(from, to, amount))
    .setRecentBlockhash(latestBlock.hash)
    .build();

  console.log(`Transfer: ${amount / 1e9} MOLT`);
  console.log(`From: ${from.toBase58()}`);
  console.log(`To: ${to.toBase58()}`);
  console.log(`Blockhash: ${latestBlock.hash.substring(0, 16)}...`);

  // Note: In a real application, you would sign the transaction here
  // For example: transaction.sign(keypair);

  console.log('\n⚠️  Transaction built but not signed or sent');
  console.log('In a real app, you would sign with a keypair and send with connection.sendTransaction()');
}

main().catch(console.error);

File: sdk/python/examples/transaction.py
"""Transaction example for MoltChain Python SDK"""

import asyncio
from moltchain import Connection, PublicKey, TransactionBuilder


async def main():
    connection = Connection('http://localhost:8899')
    
    print('🦞 MoltChain Transaction Example\n')
    
    # Create a transfer transaction
    from_pubkey = PublicKey('FromPublicKeyHere...')
    to_pubkey = PublicKey('ToPublicKeyHere...')
    amount = 1_000_000_000  # 1 MOLT
    
    print('Building transaction...')
    
    # Get recent blockhash
    latest_block = await connection.get_latest_block()
    
    # Build transaction
    message = (TransactionBuilder()
        .add(TransactionBuilder.transfer(from_pubkey, to_pubkey, amount))
        .set_recent_blockhash(latest_block['hash'])
        .build())
    
    print(f"Transfer: {amount / 1e9} MOLT")
    print(f"From: {from_pubkey}")
    print(f"To: {to_pubkey}")
    print(f"Blockhash: {latest_block['hash'][:16]}...")
    
    # Note: In a real application, you would sign the transaction here
    # For example: transaction = sign_transaction(message, keypair)
    
    print('\n⚠️  Transaction built but not signed or sent')
    print('In a real app, you would sign with a keypair and send with connection.send_transaction()')


if __name__ == '__main__':
    asyncio.run(main())
